setup.py

```diff
@@ -30,45 +32,37 @@
 # Extensions need to link against appropriate libs
 # We use pkg-config to find the appropriate set of includes and libs
 def call_package_config(package, option, optional=False):
-    '''invoke pkg-config, if it exists, to find the appropriate
-    arguments for a library'''
+    '''invoke pkg-config, if it exists, to find the appropriate arguments for a library'''
     try:
         cp = subprocess.run(["pkg-config", package, option],
                             universal_newlines=True, stdout=subprocess.PIPE)
     except FileNotFoundError:
-        print("Unable to check for %s, pkg-config not installed" %
-              package, file=sys.stderr)
+        print(f"Unable to check for '{package}', pkg-config not installed", file=sys.stderr)
         if optional:
             return []
         else:
             sys.exit(1)
 
     if cp.returncode != 0:
         if optional:
-            print("Can't find '%s'" % package, file=sys.stderr)
+            print(f"Can't find '{package}'", file=sys.stderr)
             print("Some functionality will be disabled", file=sys.stderr)
             return []
         else:
-            print("Development files not found for: '%s'." %
-                  package, file=sys.stderr)
+            print(f"Development files not found for '{package}'", file=sys.stderr)
             sys.exit(1)
 
     return cp.stdout.split()
 
 
 png_flags = call_package_config("libpng", "--cflags")
 png_libs = call_package_config("libpng", "--libs")
+define_macros = [('PNG_ENABLED', 1)]
 
-extra_macros = [('PNG_ENABLED', 1)]
-
-jpg_lib = "jpeg"
-for path in "/usr/include/jpeglib.h", "/usr/local/include/jpeglib.h":
-    if os.path.isfile(path):
-        extra_macros.append(('JPG_ENABLED', 1))
-        jpg_libs = [jpg_lib]
-        break
-else:
-    raise SystemExit("NO JPEG HEADERS FOUND, you need to install libjpeg-dev")
+jpeg_flags = call_package_config("libjpeg", "--cflags")
+jpeg_libs = call_package_config("libjpeg", "--libs")
+if jpeg_flags + jpeg_libs:
```

Comment:
call_package_config() will exit if the package isn't found. Is the concern that jpeg_libs could be empty (that would be a broken system)? In that case suggest changing to a test for `not jpeg_libs` that exits with an error rather than silently proceeding. N.B. PNG_ENABLED above as well.

If both jpeg and png are not optional (as suggested by the call_package_config with optional=False) can the macros be removed?

Reply:
Good catch,
png was already but it's my mistake making jpg mandatory too.

As it is now just a part of gnofract4d, it could make sense to remove macros for mandatory includes. Anyway if the png lib is not found it won't even try to compile fract4dc.
On the other hand, if you look at the C++ layer as something that could be reused elsewhere, it still makes sense to have them (though you could find a better approach).

Reply:
I don't mind either way whether we make jpg mandatory